src/test/common/clara/test_performance_optimizations.cljc

```diff
@@ -0,0 +1,68 @@
+;; These tests validate that operations that the rules engine should optimize
+;; away are in fact optimized away.  The target here is not the actual execution time,
+;; which will vary per system, but verification that the action operations in question are not performed.
+#?(:clj
+   (ns clara.test-performance-optimizations
```

Comment:
I think it's pretty heavy-weight to fully repeat the whole `ns` form across the 2 read conditionals.
Ideally we'd be minimizing the need to even specialize most parts of it for clj vs cljs.

For example, `:require-macros` should be unnecessary in most cases nowadays.

Another good simplification is to be pretty aggressive in the use of ns aliases instead of `:refer`. That can make it so that the need for any "<something>-macros" cljs declarations are unnecessary too.

This may not be that important for the sake of this review. I just think this will become an annoying pattern to maintain/work with.

Reply:
I can take a look at this, either as part of this PR or otherwise.  When I initially started creating these test namespaces I wasn't able to find a way to share enough to make it seem worthwhile - IIRC around 80% of the code in the header was in conditionals when I tried to make it a single header.  However I could have been missing some options in how to create CLJS ns headers.

```
Another good simplification is to be pretty aggressive in the use of ns aliases instead of :refer. That can make it so that the need for any "-macros" cljs declarations are unnecessary too.
```
How does the usage of aliases allow removal of require-macros declarations?